specs/online_order_spec.rb

```diff
@@ -3,69 +3,145 @@
+      end
+
+
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new(
+      1234,
+      { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"),
+      :processing)
+      proc{online_order.add_product("yogurt", 3.50)}.must_raise ArgumentError
+
+      online_order = Grocery::OnlineOrder.new(
+      1234,
+      { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"),
+      :shipped)
+      proc{online_order.add_product("yogurt", 3.50)}.must_raise ArgumentError
+
+      online_order = Grocery::OnlineOrder.new(
+      1234,
+      { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"),
+      :complete)
+      proc{online_order.add_product("yogurt", 3.50)}.must_raise ArgumentError
     end
 
+
     it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      online_order = Grocery::OnlineOrder.new(
+      1234,
+      { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"),
+      :pending)
+      online_order.add_product("yogurt", 3.50)
+      online_order.products.include?("yogurt").must_equal true
     end
   end
 
+
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
-      # TODO: Your test code here!
       # Useful checks might include:
       #   - OnlineOrder.all returns an array
       #   - Everything in the array is an Order
       #   - The number of orders is correct
       #   - The customer is present
       #   - The status is present
       # Feel free to split this into multiple tests if needed
+      Grocery::OnlineOrder.all do |s|
+        s.must_be_kind_of Grocery::OnlineOrder
+      end
+    end
+
+    it " The number of orders must be correct" do
+      online_order = Grocery::OnlineOrder.all
+      online_order.length.must_equal 100
+    end
+
+    it "Customer must be present" do
+      online_order = Grocery::OnlineOrder.all
+      online_order[0].customer.must_be_kind_of Grocery::Customer
+      online_order[0].customer.id.must_equal "25"
+      online_order[0].customer.email.must_equal "summer@casper.io"
+      online_order[0].customer.address.must_equal "66255 D'Amore Parkway,New Garettport,MO,57138"
+    end
+
+    it "The status must be present" do
+      online_order = Grocery::OnlineOrder.all
+      online_order[0].status.must_equal :complete
     end
   end
-  
+
   describe "OnlineOrder.find_by_customer" do
     it "Returns an array of online orders for a specific customer ID" do
-      # TODO: Your test code here!
+      order = Grocery::OnlineOrder.find_by_customer("25")
+      order.id.must_equal "1"
```

Comment:
This test claims to check that `find_by_customer` returns an *array* of `OnlineOrders`, but you instead seem to be checking that it returns a *single* `OnlineOrder`. It's important to make sure that your test name matches what the test actually does.